backend/app/anti_recommenders/arkg/arkg_anti_recommender.py

```diff
@@ -1,162 +1,209 @@
-import typing
+import logging
 from collections.abc import Iterable
 from pathlib import Path
+from typing import override
 
 import pyoxigraph as ox
+from postgrest import APIError
 
 from app.anti_recommenders import AntiRecommender
-from app.constants import WIKIPEDIA_BASE_URL
-from app.models import AntiRecommendation, AntiRecommendationGraph
+from app.constants import (
+    ARKG_ANTI_RECOMMENDER_USER_STATE_TABLE_NAME,
+    WIKIPEDIA_BASE_URL,
+)
+from app.database import database_client
+from app.models import AntiRecommendation, User, UserState
 from app.models.types import RdfMimeType, RecordKey, StoreQuery
 from app.namespaces import SCHEMA
 
 
 class ArkgAntiRecommender(AntiRecommender):
```

Comment:
Feels weird to have database queries and manipulation in an anti-recommender. The usual pattern would be to encapsulate the queries and manipulation under database interfaces. That also makes it easier to test the bit of anti-recommender logic.